lib/ABI/FunctionType.cpp

```diff
@@ -151,66 +177,148 @@ class ConversionHelper {
     for (auto &SavedRegister : AT::CalleeSavedRegisters)
       revng_assert(model::Register::getArchitecture(SavedRegister) == Arch);
 
-    model::CABIFunctionType Result;
-    Result.CustomName = Function.CustomName;
-    Result.OriginalName = Function.OriginalName;
-    Result.ABI = ABI;
+    // A single place to conveniently store all the types that need to be added
+    // to the model in order for the new function prototype to be valid.
+    NewTypeContainer TypeDependencies;
+
+    // Convert all the arguments.
+    static constexpr auto &GPAR = AT::GeneralPurposeArgumentRegisters;
+    auto RegisterArguments = tryConvertingRegisterArguments(Function.Arguments,
+                                                            GPAR,
+                                                            *TheBinary,
+                                                            TypeDependencies);
+    if (RegisterArguments == std::nullopt)
+      return std::nullopt;
 
-    if (!verifyArgumentsToBeConvertible(Function.Arguments,
-                                        AT::GeneralPurposeArgumentRegisters,
-                                        *TheBinary))
```

Comment:
It seems there's some refactoring going on, in parallel to new features, which makes it a bit difficult for me to review things.

Can you briefly point me to *new* things so I can review them ignoring the diff?

Reply:
Yeah, I'll look into splitting this commit up as soon as my working tree is a bit cleaner, sorry about that.

Reply:
I've split it. Sorry for the wait.